src/Materials/babylon.material.ts

```diff
@@ -1069,14 +1069,28 @@
                     }
                 }
 
-                if (options && options.clipPlane) {
+                if (localOptions.clipPlane) {
                     scene.clipPlane = clipPlaneState;
                 }
             };
 
             checkReady();
         }
 
+        /**
+         * Force shader compilation.
+         * @param mesh The mesh that will use this material
+         * @param options Additional options for compiling the shaders
+         * @returns A promise that resolves when the compilation completes
+         */
+        public forceCompilationAsync(mesh: AbstractMesh, options?: Partial<{ clipPlane: boolean }>): Promise<void> {
+            return new Promise(resolve => {

```

Comment:
Should we check  compilation error ?

Reply:
Not sure. It didn't before. If we think this should happen, let's do it in a separate change.